packages/netsuite-adapter/src/change_validators/immutable_changes.ts

```
@@ -44,39 +44,64 @@ const toModifiedAnnotationChangeError = (
     + 'In order to proceed with this deployment, please edit the element in Salto and remove this annotation change.',
 })
 
-const toTypeErrors = async (change: ModificationChange<ObjectType>): Promise<ChangeError[]> => {
-  const { before, after } = change.data
-  const modifiedImmutableAnnotations = await awu(Object.entries(after.annotationRefTypes))
+const typeServiceIdConditions = <T extends (AdditionChange<ObjectType> | ModificationChange<ObjectType>)>(
+  change: T,
+  condition: (change: T, annoName: string) => boolean
+): Promise<string[]> => {
+  const { after } = change.data
+  return awu(Object.entries(after.annotationRefTypes))
```

There can be a case where the service id annotation refType doesn't exist too. In this case we won't catch a missing service id annotation. I think that we should return an error in case that no annotation is the service id (specifically in types).